### PR TITLE
Update <Site> component to use ES6 syntax.

### PR DESCRIPTION
**Pull request by mtias**

### CODE REVIEW

File: client/my-sites/site/index.jsx [hunk 1]
```diff
@@ -1,26 +1,27 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' ),
-	noop = require( 'lodash/utility/noop' );
+import React from 'react';
+import classnames from 'classnames';
```
Comment (retrofox) [inline client/my-sites/site/index.jsx]: are you avoiding camelCase on purpose?

**Comment by retrofox** [comment]
BTW I got these eslint warning 

image

it comes from [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md) module.

My suggestion:

```es6
const setStartToottip = startTooltip => this.setState( { startTooltip } );

return (
  <button
    className="site__star"
    onClick={ this.starSite }
    onMouseEnter={ setStartTooltip( true ) }
    onMouseLeave={ setStartTooltip( false ) }
    // ...
```

File: client/my-sites/site/index.jsx (hunk 1 above)
Comment (mtias) [inline client/my-sites/site/index.jsx]: I don't think it's necessary here.

**Comment by gwwar** [comment]
If we're bothering to work around the warning I'd suggest doing it properly. Basically we don't want to create functions in a render method. One way to fix this would be to add:
```javascript
enableStarTooltip() {
    this.setState( { starTooltip: true } );
},
disableStarTooltip() {
    this.setState( { starTooltip: false } );
},
render() {
    //...
   <button onMouseEnter={ this.enableStarTooltip } onMouseLeave={ this.disableStarTooltip } />
}
```

**Comment by gwwar** [comment]
:+1: changes look good to :ship: after a rebase

**Comment by blowery** [comment]
@gwwar Marking this back as Needs Review since I rebased it and added the bound methods to get rid of the JSX-bind linting warnings.

File: client/my-sites/site/index.jsx [hunk 2]
```diff
@@ -75,12 +76,20 @@ module.exports = React.createClass( {
...
 
-	starSite: function() {
+	starSite() {
 		const site = this.props.site;
 		sites.toggleStarred( site.ID );
 	},
 
-	renderStar: function() {
+	enableStarTooltip() {
+		this.setState( { starTooltip: true } )
```
Comment (gwwar) [inline client/my-sites/site/index.jsx]: ; ?

File: client/my-sites/site/index.jsx [hunk 3]
```diff
@@ -75,12 +76,20 @@ module.exports = React.createClass( {
...
+	},
+
+	disableStarTooltip() {
+		this.setState( { starTooltip: false } )
```
Comment (gwwar) [inline client/my-sites/site/index.jsx]: ; ?

**Comment by gwwar** [comment]
:ship: after adding in the missing ;

**Comment by blowery** [comment]
@gwwar nice catch! fixed in an amended commit.

**Pull request merged**